src/shrinkray/problem.py

```diff
@@ -162,6 +162,8 @@ class ReductionProblem[T](ABC):
     """
 
     work: WorkContext
+    # Track current pass stats for real-time updates (set by reducer)
+    current_pass_stats: Any = None
```

Comment:
Why is this an Any?

(Any is not flat out forbidden, but you should think hard about whether Any is needed in any given location)